exporter.go

```diff
@@ -371,16 +371,22 @@ func (e *Exporter) entrypoint(launchMD launch.Metadata, defaultProcessType strin
 	if !e.supportsMulticallLauncher() {
 		return launch.LauncherPath, nil
 	}
-	if defaultProcessType == "" {
-		if len(launchMD.Processes) == 1 {
-			e.Logger.Infof("Setting default process type '%s'", launchMD.Processes[0].Type)
-			return launch.ProcessPath(launchMD.Processes[0].Type), nil
+
+	if defaultProcessType != "" {
+		defaultProcess, ok := launchMD.FindProcessType(defaultProcessType)
+		if !ok {
+			if e.PlatformAPI.Compare(api.MustParse("0.6")) < 0 {
+				e.Logger.Warn(processTypeWarning(launchMD, defaultProcessType))
+				return launch.LauncherPath, nil
+			}
+			return "", fmt.Errorf("tried to set %s to default but it doesn't exist", defaultProcessType)
```

Comment:
Elsewhere in this file we're using a pattern of `errors.Wrap(err, "do a thing")`

Reply:
There are also some places in this file that we're using `fmt.Errorf` (for example: https://github.com/buildpacks/lifecycle/blob/fae1fe27c0acdfabe9538e3607e72af61c62ede9/exporter.go#L431).
I think it's better to use `fmt.Errorf` in this case since we don't have an error to wrap.
What do you think?

Reply:
Ohh I missed that!